dev/tools/pre-commit

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# Copy to .git/hooks/ to use.
+
+set -e
+
+# NB: even though we use --cached it seems to exit code 1 with
+# unstaged whitespace errors. That just means they get fixed though.
+if git diff-index --check --cached HEAD --quiet;
```

Comment:
I think this is the wrong check.  `--quiet` implies `--exit-code`, which means that this command fails when there are any changes whatsoever.  So this is basically just a check that the commit is not empty.  The whitespace fixer is fast enough that this doesn't really hurt, though.

Reply:
>I think this is the wrong check. --quiet implies --exit-code, which means that this command fails when there are any changes whatsoever. So this is basically just a check that the commit is not empty.

I guess that explains what I saw in the comment.